tests/unit/s2n_resume_test.c

```diff
@@ -211,7 +211,7 @@ int main(int argc, char **argv)
             struct s2n_connection *conn = s2n_connection_new(S2N_CLIENT);
             EXPECT_NOT_NULL(conn);
 
-            EXPECT_OK(s2n_client_deserialize_session_state(conn, &ticket_stuffer));
+            EXPECT_OK(s2n_client_deserialize_session_state(conn, NULL, &ticket_stuffer));
```

Comment:
The existing unit tests only check for client side. Can we add unit tests to sanity check the server side  deserialize logic as well?